doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -1065,11 +1065,12 @@ use cases of this functionality is the association of Context Providers.
 
 A context registration is represented by a JSON object with the following fields:
 
-+ `id`: Unique identifier assigned to the registration. Automatically created at creation time.
-+ `subject` : It s an object that describes the subject of the registration.
-+ `callback` : URL pointing to the service which is registered. In the case of a Context Provider
-corresponds to the URL of the provider service.
-+ `duration`: Duration of the registration in ISO8601 format. Default duration is infinite.
++ `id`       : Unique identifier assigned to the registration. Automatically generated at creation time.
++ `subject`  : It s an object that describes the subject of the registration.
++ `callback` : URL which denotes the end point of the registered service. It can be undefined under certain conditions,
+for instance, transient unavailability of the provider.
++ `metadata` : An optional dictionary which allows to associate extra metadata to the registration. 
```

Comment:
It depends on whether we want to keep entities and registration metadata coherent (as FIWARE NGSIv1 and NGSI OMA do) or not. In my opinion, coherence is important (otherwise, users of the API has to remember that entity metadata use one approach, registration metadata use a different approach), so I would prefer that option.

However, maybe @tobiasjacobs is the one that could provide a better opinion here, as the supporter of the use case of metadata for registration is him :)